ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -89,6 +89,7 @@
   private static final Set<String> IMMUTABLE_PROPERTIES = ImmutableSet.<String>builder()
       .add(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG)
       .add(KsqlConfig.KSQL_EXT_DIR)
+      .add(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG)
```

Comment:
We don't need this in this list, since the config isn't used by the engine.

Reply:
This list is checked from buildMultipleQueries(). Is that not where we want to enforce that this property cannot be overridden?

Reply:
We don't care about the property from buildMultipleQueries right? The engine doesn't use it, its enforced by the server from StatementExecutor, which doesn't consider the user's overrides anyway.

Reply:
Good point about the property being irrelevant to buildMultipleQueries, and StatementExecutor not considering user overrides. However, it turns out that when LocalPropertyValidator is instantiated (from LocalPropertyParser), the set of immutable properties it checks for is `KsqlEngine.getImmutableProperties()`, so leaving this the query limit config in this list of immutable properties seems correct after all. (Funny how things work out!)

Reply:
I think it makes more sense to do something like:

```
LocalPropertyValidator() {
    this(
        new ImmutableList.Builder<String>()
            .addAll(ksqlEngine.getImmutableProperties())
            .add(KsqlConfig. KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG)  // (or something like  StatementExecutor.getImmutableProperties())
            .build);
}
}
```

Reply:
Created: https://github.com/confluentinc/ksql/issues/2154.